tests/csrf_tests/tests.py

```diff
@@ -707,6 +703,17 @@ def _set_post(self, post):
             'Forbidden (%s): ' % REASON_CSRF_TOKEN_MISSING,
         )
 
+    def test_post_data_local_storage_error(self):
+        """
+        Unexpected OSError during storing files should end up in a 500 response.
+        """
+        token = ('ABC' + self._csrf_id_token)[:CSRF_TOKEN_LENGTH]
+        mw = CsrfViewMiddleware(post_form_view)
+        req = CsrfPostRequest(token, error_to_raise=OSError('Deleted directories/Missing permissions'))
```

Comment:
If this request class is going to be factored out, I think it would be better to add a `request_class` argument to `_get_POST_csrf_cookie_request()` and call that method. That way the `CSRF_USE_SESSIONS` true / false code isn't duplicated, as well as simultaneously included. `_get_POST_csrf_cookie_request()` should handle that automatically, using the overridden methods in the concrete test class being run. The exception to raise can be a class attribute.